src/ra_deleters.h

```diff
@@ -0,0 +1,211 @@
+                ::FindClose(hObject);
+                break;
+            case NTKernelObjType::TimerQueue:
+                ::DeleteTimerQueue(hObject);
+                break;
+            case NTKernelObjType::File:     [[fallthrough]]; // all the way to Object
+            case NTKernelObjType::Job:      [[fallthrough]];
+            case NTKernelObjType::Mailslot: [[fallthrough]];
+            case NTKernelObjType::Mutex:    [[fallthrough]];
+            case NTKernelObjType::Pipe:     [[fallthrough]];
+            case NTKernelObjType::Process:  [[fallthrough]];
+            case NTKernelObjType::Thread:   [[fallthrough]];
+            case NTKernelObjType::Object: /*tags above are meant for type aliasing but most aren't used right now */
+                ::CloseHandle(hObject);
+        }
+    }
+};
+
+} /* namespace detail */
+
+// For these types all you need to do is specify the creation function
+using CFileH    = std::unique_ptr<FILE, detail::CFileDeleter>; // there's an HFILE as well so we put CFile
+using MenuH     = std::unique_ptr<HMENU__, detail::MenuDeleter>;
+using InstanceH = std::unique_ptr<HINSTANCE__, detail::InstanceDeleter>;
+using AccelH    = std::unique_ptr<HACCEL__, detail::AcceleratorDeleter>;
+using DCH       = std::unique_ptr<HDC__, detail::DcDeleter>; // Do not use with a Window DC
+using InternetH = std::unique_ptr<void, detail::InternetDeleter>;
+using GlobalH   = std::unique_ptr<void, detail::GlobalDeleter>;
+
+// Only for use if HICON is initialized with CreateIconFromResourceEx (if called without the LR_SHARED flag),
+// CreateIconIndirect, and CopyIcon. HCURSOR cannot be shared either
+using IconH   = std::unique_ptr<HICON__, detail::IconDeleter<detail::IconHandle::Icon>>;
+using CursorH = std::unique_ptr<HICON__, detail::IconDeleter<detail::IconHandle::Cursor>>;
+
+using AppWndH      = std::unique_ptr<HWND__, detail::WindowDeleter<detail::WindowType::App>>;
+using ModelessDlgH = std::unique_ptr<HWND__, detail::WindowDeleter<detail::WindowType::ModelessDialog>>;
+using ModalDlgH    = std::unique_ptr<HWND__, detail::WindowDeleter<detail::WindowType::ModalDialog>>;
+
+using BitmapH  = std::unique_ptr<HBITMAP__, detail::GdiDeleter<detail::GdiType::Bitmap>>;
+using BrushH   = std::unique_ptr<HBRUSH__, detail::GdiDeleter<detail::GdiType::Brush>>;
+using FontH    = std::unique_ptr<HFONT__, detail::GdiDeleter<detail::GdiType::Font>>;
+using PaletteH = std::unique_ptr<HPALETTE__, detail::GdiDeleter<detail::GdiType::Palette>>;
+using RgnH     = std::unique_ptr<HRGN__, detail::GdiDeleter<detail::GdiType::Rgn>>;
+using GdiObjH  = std::unique_ptr<void, detail::GdiDeleter<detail::GdiType::Obj>>;
+
+using FileH     = std::unique_ptr<void, detail::NTKernelDeleter<detail::NTKernelObjType::File>>;
+using FindFileH = std::unique_ptr<void, detail::NTKernelDeleter<detail::NTKernelObjType::FindFile>>;
+using HandleH   = std::unique_ptr<void, detail::NTKernelDeleter<detail::NTKernelObjType::Object>>;
```

Comment:
Typing out the right side each time and specifying the deleter each time on construction would be tedious. These type aliases attempt to make it simpler.